src/components/calcite-input/calcite-input.tsx

```diff
@@ -284,6 +287,7 @@ export class CalciteInput {
         this.value = undefined;
       }
     }
+    this.effectiveLabel = this.label || findFormFieldLabel(this.el);
```

Comment:
We may want to do this on render in case the labeling elements aren't added to the DOM yet.